src/commands/part.cpp

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2024 <faboussa>************************************************* */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   part.cpp                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: faboussa <faboussa@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/17 11:50:56 by faboussa          #+#    #+#             */
+/*   Updated: 2024/11/08 17:47:18 by faboussa         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+// ajout fonction is client in channel
+
+#include <map>
+#include <stdexcept>
+#include <string>
+
+#include "../includes/Client.hpp"
+#include "../includes/Server.hpp"
+#include "../includes/colors.hpp"
+#include "../includes/numericReplies.hpp"
+#include "../includes/utils.hpp"
+
+void Server::quitChannel(int fd) {
+  Client *client = &_clients.at(fd);
+
+  channelsMap::iterator itEnd = _channels.end();
+  channelsMap::iterator itBegin = _channels.begin();
+  for (channelsMap::iterator it = itBegin; it != itEnd; ++it) {
```

Comment:
Removed `channelsMap::iterator itBegin = _channels.begin();` and directly initialize `it` at the beginning of for loop.
```
channelsMap::iterator itEnd = _channels.end();
  for (channelsMap::iterator it = _channels.begin(); it != itEnd; ++it) 
```